newIDE/app/src/MainFrame/EditorTabs/EditorTabsHandler.js

```diff
@@ -289,3 +289,24 @@ export const getEventsFunctionsExtensionEditor = (
 
   return null;
 };
+
+export const moveTabToPosition = (
+  editorTabsState: EditorTabsState,
+  fromIndex: number,
+  toIndex: number
+): EditorTabsState => {
+  if (toIndex < fromIndex) toIndex += 1;
+  const currentEditorTabs = [...getEditors(editorTabsState)];
+  const movingTab = currentEditorTabs[fromIndex];
+  currentEditorTabs.splice(fromIndex, 1);
+  currentEditorTabs.splice(toIndex, 0, movingTab);
+
+  let currentTabIndex: number = getCurrentTabIndex(editorTabsState);
+  if (fromIndex === currentTabIndex) currentTabIndex = toIndex;
```

Comment:
Can you give a name to those booleans to improve readability?
This one could be named
```
const theMovingTabIsTheCurrentTab = fromIndex === currentTabIndex
```

And, as a suggestion:

```
const movingTabWentFromLeftOfCurrentTabToRight = fromIndex < currentTabIndex && toIndex >= currentTabIndex
```